### DISCUSSION TITLE
Pre-configured settings from file

### DISCUSSION THREAD

**Discussion by Pt-PhilW** [foxyproxy/firefox-extension#191] [Q&A]
Is it possible to import settings from a file? My goal is to have a pre-configured proxy that can be used without having to first import settings by using the import-function in the browser or adding a proxy manually.
If so where does the settings-file need to be put and what is the required structure of the file?

**Comment by erosman** [comment]
> Is it possible to import settings from a file? My goal is to have a pre-configured proxy that can be used without having to first import settings by using the import-function in the browser or adding a proxy manually.

How and where from FoxyProxy is going to get those settings without user-action?
Local file access is not possible in Firefox.

The import option requires user-action i.e. clicking button to import it.

There would be a possibility in [FoxyProxy v8.0](https://github.com/foxyproxy/browser-extension) (once released) to set FoxyProxy to use an external PAC.

That way, user can create a PAC and place it somewhere that is accessible via HTTP/s.
